WcaOnRails/app/views/competitions/_index_form.html.erb

```diff
@@ -35,21 +35,29 @@
 </div>
 
 <div class="form-group">
-  <%= label_tag(:state) %>
+  <%= label_tag(t('.state')) %>
   <div id="state" class="btn-group" data-toggle="buttons">
     <label id="present" class="btn btn-primary">
       <%= radio_button_tag "state", "present" %>
-      <span class="caption">Present</span>
+      <span class="caption"><%= t '.present' %></span>
     </label>
     <ul class="dropdown-menu years">
       <%= hidden_field_tag "year", params[:year] %>
       <% @years.each do |year| %>
-        <li class="year"><%= link_to year.to_s.capitalize, "#", data: { year: year } %></li>
+        <% year_label = year.is_a?(Integer) ? year.to_s : t('.all_years') %>
+        <li class="year"><%= link_to year_label, "#", data: { year: year } %></li>
       <% end %>
     </ul>
     <label id="past" class="btn btn-primary" data-toggle="dropdown">
       <%= radio_button_tag "state", "past" %>
-      <span class="caption">Past <%= "from #{params[:year]}" if @past_selected %></span>
+      <span class="caption">
+        <% if @past_selected %>
+          <% #Here params[:year] is a string, so we need regexp %>
+          <%= params[:year] =~ /\A\d+\z/ ? t('.past_from', year: params[:year]) : t('.past_all') %>
```

Comment:
Here is an url generated by the page in French:

> http://localhost:3000/competitions?utf8=%E2%9C%93&region=all&search=&year=all+years&state=past&display=list

Parameters' names and values are not changed depending on the locale, only the captions and labels are :) (unless I did something wrong)
Here this statement means "if params[:year] matches a positive integer, use the 'past_from' key, else use the 'past_all' key".

Does it sound good?